contracts/hydro/src/contract.rs

```diff
@@ -1295,6 +1328,129 @@ fn withdraw_icq_funds(
+// Having more arguments rather than nested structures actually makes the json easier to construct when calling this from the outside,
+// so we allow many arguments to this function.
+#[allow(clippy::too_many_arguments)]
+pub fn add_liquidity_deployment(
+    deps: DepsMut<NeutronQuery>,
+    env: Env,
+    info: MessageInfo,
+    round_id: u64,
+    tranche_id: u64,
+    proposal_id: u64,
+    destinations: Vec<String>,
+    deployed_funds: Vec<Coin>,
+    funds_before_deployment: Vec<Coin>,
+    total_rounds: u64,
+    remaining_rounds: u64,
+) -> Result<Response<NeutronMsg>, ContractError> {
+    let constants = CONSTANTS.load(deps.storage)?;
+
+    validate_contract_is_not_paused(&constants)?;
+    validate_sender_is_whitelist_admin(&deps, &info)?;
+
+    // check that the round has started
+    let current_round_id = compute_current_round_id(&env, &constants)?;
+    if round_id > current_round_id {
+        return Err(ContractError::Std(StdError::generic_err(
+            "Cannot add liquidity deployment for a round that has not started yet",
+        )));
+    }
+
+    // check that the tranche with the given id exists
+    TRANCHE_MAP.load(deps.storage, tranche_id)?;
+
+    // check that the proposal with the given id exists
+    PROPOSAL_MAP
+        .load(deps.storage, (round_id, tranche_id, proposal_id))
+        .map_err(|_| {
+            ContractError::Std(StdError::generic_err(format!(
+                "Proposal for round {}, tranche {}, and id {} does not exist",
+                round_id, tranche_id, proposal_id
+            )))
+        })?;
+
+    // check that there is no deployment for the given round, tranche, and proposal
+    if LIQUIDITY_DEPLOYMENTS_MAP
+        .may_load(deps.storage, (round_id, tranche_id, proposal_id))?
+        .is_some()
+    {
+        return Err(ContractError::Std(StdError::generic_err(
+            "There already is a deployment for the given round, tranche, and proposal",
+        )));
+    }
+
+    let deployment = LiquidityDeployment {
+        round_id,
+        tranche_id,
+        proposal_id,
+        destinations,
+        deployed_funds,
+        funds_before_deployment,
+        total_rounds,
+        remaining_rounds,
+    };
+
+    let response = Response::new()
+        .add_attribute("action", "add_round_liquidity_deployments")
+        .add_attribute("sender", info.sender)
+        .add_attribute("round_id", round_id.to_string())
+        .add_attribute("tranche_id", tranche_id.to_string())
+        .add_attribute("proposal_id", proposal_id.to_string())
+        .add_attribute(
+            "deployment",
+            serde_json_wasm::to_string(&deployment).map_err(|_| {
+                ContractError::Std(StdError::generic_err("Failed to serialize deployment"))
+            })?,
+        );
+
+    LIQUIDITY_DEPLOYMENTS_MAP.save(
+        deps.storage,
+        (round_id, tranche_id, proposal_id),
+        &deployment,
```

Comment:
Just checking: the inputs are not validated. Since the function can be called by the trusted 3rd party, maybe we don't need to validate them?